include/seqan3/argument_parser/validators.hpp

```diff
@@ -352,6 +353,39 @@ class file_ext_validator // TODO AFTER INITIAL MERGE: Check if file exists, also
     std::vector<std::string> extensions;
 };
 
+/*!\brief A validator that checks if a file exists.
+ * \ingroup argument_parser
+ *
+ * \details
+ *
+ * The struct then acts as a functor that throws a seqan3::parser_invalid_argument
+ * exception whenever a given filename (string) does not exist.
+ *
+ * \snippet test/snippet/argument_parser/validators_3.cpp usage
+ */
+class file_existance_validator
+{
+public:
+    //!\brief Type of values that are tested by validator
+    using value_type = std::string;
```

Comment:
This needs to be `filesystem::path` I think! When you create the snippet I mentioned before this could otherwise be a source of a compiler error.